### PR TITLE
AC Automated Test - LRAC-9101, LRAC-9106, LRAC-9109 | master

### PR DESCRIPTION
**Pull request by LucasAcacio**
https://issues.liferay.com/browse/LRAC-9101
https://issues.liferay.com/browse/LRAC-9106
https://issues.liferay.com/browse/LRAC-9109

### CODE REVIEW

**Comment by LucasAcacio** [comment]
ci:test:sf

File: portal-web/test/functional/com/liferay/portalweb/macros/analyticscloud/ACProperties.macro [hunk 1]
```diff
@@ -55,6 +54,12 @@ definition {
 			value1 = "${propertyName}");
 	}
 
+	@summary = "Adds new property"
+
```
Comment (alexcandidos) [inline portal-web/test/functional/com/liferay/portalweb/macros/analyticscloud/ACProperties.macro:58]: hi @LucasAcacio  this summary is not related to this macro. Could you please double-check?

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/Properties.testcase [hunk 2]
```diff
@@ -417,21 +424,41 @@ definition {
...
+		task ("Add a new property with the minimum char quantity") {
+			ACProperties.addNewProperty(propertyName = "Min");
+		}
 
-		// TODO LRAC-9106 PropertyNameWithCharQuantityValidation pending implementation
+		task ("Add a new property with the maximum char quantity") {
+			ACSettings.goToProperties();
 
+			ACProperties.addNewProperty(propertyName = "MaximumMaximumMaximumMaximumMaximumMaximumMaximumMaximumMaximumMa");
+		}
```
Comment (alexcandidos) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/Properties.testcase:461]: hi @LucasAcacio , I think this case is a boundary test. So, you should investigate which is the minium char quantity and try to assert you are not able to create a property with minimum -1/+1, validating any alert info that is shown, that save button is disabled, and the same for maximum + 1; 

What do you think? Thanks!

Comment (LucasAcacio) [reply portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/Properties.testcase:461]: Sounds good!

**Comment by LucasAcacio** [comment]
ci:test:sf

File: portal-web/test/functional/com/liferay/portalweb/macros/analyticscloud/ACProperties.macro [hunk 3]
```diff
@@ -40,6 +40,18 @@ definition {
...
+	macro assertCannotAddProperty {
+		Click(locator1 = "ACSettings#NEW_PROPERTY_BUTTON");
+
+		Type(
+			locator1 = "ACSettings#PROPERTY_INPUT",
+			value1 = "${propertyName}");
+
+		AssertElementPresent(locator1 = "AppBuilder#SAVE_BUTTON_DISABLED");
+
+		Click(locator1 = "TableViewBuilder#CLEAR_SEARCH");
```
Comment (alexcandidos) [inline portal-web/test/functional/com/liferay/portalweb/macros/analyticscloud/ACProperties.macro:52]: hi @LucasAcacio , this macro is doing more actions than needed.

I would suggest you keep these actions in the test case body:

```
		Click(locator1 = "ACSettings#NEW_PROPERTY_BUTTON");

		Type(
			locator1 = "ACSettings#PROPERTY_INPUT",
			value1 = "${propertyName}");

	        Click(locator1 = "TableViewBuilder#CLEAR_SEARCH");
```

and only this assert on the macro:
`AssertElementPresent(locator1 = "AppBuilder#SAVE_BUTTON_DISABLED");`

**Comment by LucasAcacio** [comment]
ci:test:sf

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/Properties.testcase [hunk 4]
```diff
@@ -526,21 +533,65 @@ definition {
...
+
+			ACNavigation.goToSettings();
 
-		// TODO LRAC-9109 PropertyFieldDisplayedWhenNoPropertyAvailable pending implementation
+			ACSettings.goToProperties();
+		}
 
+		task ("assert empty state on properties page") {
+			ACProperties.assertNoProperties();
+		}
```
Comment (alexcandidos) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/Properties.testcase:548]: hi @LucasAcacio this test will fail if there are properties available.

**Comment by LucasAcacio** [comment]
ci:test:sf

**Comment by alexcandidos** [comment]
Hi @LucasAcacio , Thanks for working on it!

The code looks good to me!

I'm forwarding it.

**Comment by alexcandidos** [comment]
ci:forward
